Gems/PhysX/Code/Source/Scene/PhysXScene.cpp

```diff
@@ -974,27 +974,29 @@ namespace PhysX
         // Query flags.
         const physx::PxQueryFlags queryFlags = SceneQueryHelpers::GetPxQueryFlags(request->m_queryType);
         const physx::PxQueryFilterData queryData(queryFlags);
-
-        if (azrtti_istypeof<AzPhysics::RayCastRequest>(request))
-        {
-            return Internal::RayCast(azdynamic_cast<const AzPhysics::RayCastRequest*>(request),
-                s_rayCastBuffer, m_pxScene, queryData, m_raycastBufferSize);
-        }
-        else if (azrtti_istypeof<AzPhysics::ShapeCastRequest>(request))
-        {
-            return Internal::ShapeCast(azdynamic_cast<const AzPhysics::ShapeCastRequest*>(request),
-                s_sweepBuffer, m_pxScene, queryData, m_shapecastBufferSize);
-        }
-        else if (azrtti_istypeof<AzPhysics::OverlapRequest>(request))
+        switch (request->m_requestType)
         {
-            return Internal::OverlapQuery(azdynamic_cast<const AzPhysics::OverlapRequest*>(request),
-                s_overlapBuffer, m_pxScene, queryData, m_overlapBufferSize);
-        }
-        else
-        {
-            AZ_Warning("Physx", false, "Unknown Scene Query request type.");
-        }
-        return AzPhysics::SceneQueryHits();
+        case AzPhysics::SceneQueryRequest::Raycast:
+            {
+                return Internal::RayCast(
+                    static_cast<const AzPhysics::RayCastRequest*>(request), s_rayCastBuffer, m_pxScene, queryData, m_raycastBufferSize);
+            }
+        case AzPhysics::SceneQueryRequest::Shapecast:
+            {
+                return Internal::ShapeCast(
+                    static_cast<const AzPhysics::ShapeCastRequest*>(request), s_sweepBuffer, m_pxScene, queryData, m_shapecastBufferSize);
+            }
+        case AzPhysics::SceneQueryRequest::Overlap:
+            {
+                return Internal::OverlapQuery(
+                    static_cast<const AzPhysics::OverlapRequest*>(request), s_overlapBuffer, m_pxScene, queryData, m_overlapBufferSize);
```

Comment:
question: Do we definitely need these casts? Will things not compile without them?

Reply:
Since we're casting from base class pointer to a derived one, have to have the casts..

Reply:
Ah I see now, got you thanks